New Schedule Scripts Failing

Hey there,

I've used the old deploy scripts with no issues in the past, was trying out the automation scripts and am getting some errors:


---
> adventure-party@1.0.0 display-status
> npx hardhat run scripts/status.ts

Creating Typechain artifacts in directory typechain for target ethers-v5
Successfully generated Typechain artifacts!

/Users/Downloads/adventure-party/node_modules/ts-node/src/index.ts:692
    return new TSError(diagnosticText, diagnosticCodes);
           ^
TSError: ⨯ Unable to compile TypeScript:
typechain/factories/AdventureParty__factory.ts:16:3 - error TS2416: Property 'deploy' in type 'AdventureParty__factory' is not assignable to the same property in base type 'ContractFactory'.
  Type '(overrides?: Overrides | undefined) => Promise<AdventureParty>' is not assignable to type '(...args: any[]) => Promise<Contract>'.
    Type 'Promise<AdventureParty>' is not assignable to type 'Promise<Contract>'.
      Type 'AdventureParty' is not assignable to type 'Contract'.
        The types returned by 'queryFilter(...)' are incompatible between these types.
          Type 'Promise<TypedEvent<unknown>[]>' is not assignable to type 'Promise<Event[]>'.
            Type 'TypedEvent<unknown>[]' is not assignable to type 'Event[]'.
              Type 'TypedEvent<unknown>' is not assignable to type 'Event'.
                Types of property 'args' are incompatible.
                  Type 'unknown' is not assignable to type 'Result | undefined'.
                    Type 'unknown' is not assignable to type 'Result'.

16   deploy(overrides?: Overrides): Promise<AdventureParty> {
     ~~~~~~
typechain/factories/AdventureParty__factory.ts:22:3 - error TS2416: Property 'attach' in type 'AdventureParty__factory' is not assignable to the same property in base type 'ContractFactory'.
  Type '(address: string) => AdventureParty' is not assignable to type '(address: string) => Contract'.
    Type 'AdventureParty' is not assignable to type 'Contract'.

22   attach(address: string): AdventureParty {
     ~~~~~~
typechain/factories/AdventureParty__factory.ts:25:3 - error TS2416: Property 'connect' in type 'AdventureParty__factory' is not assignable to the same property in base type 'ContractFactory'.
  Type '(signer: Signer) => AdventureParty__factory' is not assignable to type '(signer: Signer) => ContractFactory'.
    Call signature return types 'AdventureParty__factory' and 'ContractFactory' are incompatible.
      The types of 'deploy' are incompatible between these types.
        Type '(overrides?: Overrides | undefined) => Promise<AdventureParty>' is not assignable to type '(...args: any[]) => Promise<Contract>'.

25   connect(signer: Signer): AdventureParty__factory {
     ~~~~~~~

    at createTSError (/Users/Downloads/adventure-party/node_modules/ts-node/src/index.ts:692:12)
    at reportTSError (/Users/Downloads/adventure-party/node_modules/ts-node/src/index.ts:696:19)
    at getOutput (/Users/Downloads/adventure-party/node_modules/ts-node/src/index.ts:883:36)
    at Object.compile (/Users/Downloads/adventure-party/node_modules/ts-node/src/index.ts:1185:30)
    at Module.m._compile (/Users/Downloads/adventure-party/node_modules/ts-node/src/index.ts:1309:30)
    at Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/Downloads/adventure-party/node_modules/ts-node/src/index.ts:1313:12)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at Module.require (internal/modules/cjs/loader.js:961:19)

---

any idea how to resolve?  Double checked and followed the instructions in the readme.

I deleted the artifacts folder and tried running it and had the same error.  

Actually I forgot to mention before that the new 2 lines here didn't work for me:

npm run compile
npm run test

I get:

> adventure-party@1.0.0 compile
> npx hardhat run compile

Error HH601: Script compile doesn't exist.

And:

> adventure-party@1.0.0 test
> npx hardhat run test

Creating Typechain artifacts in directory typechain for target ethers-v5
Successfully generated Typechain artifacts!
Error: Cannot find module '/Users/Downloads/adventure-party/test'

I didn't see a compile script anywhere in the repo either.

Cool so compile and test both work now, but I get the same error when running display status even after deleting the artifacts directory:

adventure-party % npm run display-status

> adventure-party@1.0.0 display-status
> npx hardhat run scripts/status.ts

Compiling 5 files with 0.8.7
Compilation finished successfully
Creating Typechain artifacts in directory typechain for target ethers-v5
Successfully generated Typechain artifacts!

/Users/Downloads/adventure-party/node_modules/ts-node/src/index.ts:692
    return new TSError(diagnosticText, diagnosticCodes);
           ^
TSError: ⨯ Unable to compile TypeScript:
typechain/factories/AdventureParty__factory.ts:16:3 - error TS2416: Property 'deploy' in type 'AdventureParty__factory' is not assignable to the same property in base type 'ContractFactory'.
  Type '(overrides?: Overrides | undefined) => Promise<AdventureParty>' is not assignable to type '(...args: any[]) => Promise<Contract>'.
    Type 'Promise<AdventureParty>' is not assignable to type 'Promise<Contract>'.
      Type 'AdventureParty' is not assignable to type 'Contract'.
        The types returned by 'queryFilter(...)' are incompatible between these types.
          Type 'Promise<TypedEvent<unknown>[]>' is not assignable to type 'Promise<Event[]>'.
            Type 'TypedEvent<unknown>[]' is not assignable to type 'Event[]'.
              Type 'TypedEvent<unknown>' is not assignable to type 'Event'.
                Types of property 'args' are incompatible.
                  Type 'unknown' is not assignable to type 'Result | undefined'.
                    Type 'unknown' is not assignable to type 'Result'.

--- a/package.json
+++ b/package.json
@@ -4,12 +4,12 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "compile": "npx hardhat run compile",
-    "test": "npx hardhat run test",
-    "deploy-party": "npx hardhat run scripts/deploy.ts",
-    "display-status": "npx hardhat run scripts/status.ts",
-    "run-party": "npx hardhat run scripts/run.ts",
-    "schedule-party": "npx hardhat run scripts/schedule.ts"
+    "compile": "hardhat compile",
+    "test": "hardhat test",
+    "deploy-party": "hardhat run --network fantom scripts/deploy.ts",
+    "display-status": "hardhat run --network fantom scripts/status.ts",
+    "run-party": "hardhat run --network fantom scripts/run.ts",
+    "schedule-party": "hardhat run --network fantom scripts/schedule.ts"
   },
   "keywords": [],
   "author": "",

--- a/scripts/common.ts
+++ b/scripts/common.ts
@@ -1,6 +1,6 @@
 import {BigNumber} from "ethers";
 import {ethers} from "hardhat";
-import { AdventureParty, AdventureParty__factory, Rarity, Rarity__factory } from "../typechain";
+import { AdventureParty, Rarity } from "../typechain";
 
 const XP_PER_DAY = BigNumber.from(250).mul(BigNumber.from(10).pow(18));
 const RARITY_CONTRACT = "0xce761d788df608bd21bdd59d6f4b54b2e27f25bb";
@@ -9,19 +9,19 @@ export async function displayStatus() {
     const [wallet] = await ethers.getSigners();
     const adventurePartyAddress = process.env.ADVENTURE_PARTY;
     if (adventurePartyAddress) {
-        const adventureParty: AdventureParty = await AdventureParty__factory.connect(adventurePartyAddress, wallet);
-        const rarity: Rarity = await Rarity__factory.connect(RARITY_CONTRACT, wallet);
+        const adventureParty = <AdventureParty> await ethers.getContractAt("AdventureParty", adventurePartyAddress);
+        const rarity = <Rarity> await ethers.getContractAt("Rarity", RARITY_CONTRACT);
 
-        const adventurersCount = await adventureParty.adventurerCount();
+        const adventurersCount = await adventureParty.connect(wallet).adventurerCount();
 
         console.log("my Adventurers: ");
 
         for (let i = 0; i < adventurersCount.toNumber(); i++) {
-            const id = await adventureParty.adventurerAt(i);
+            const id = await adventureParty.connect(wallet).adventurerAt(i);
 
-            const adventurer = await rarity.summoner(id);
-            const xpRequired = await rarity.xp_required(adventurer._level);
-            const className = await rarity.classes(adventurer._class);
+            const adventurer = await rarity.connect(wallet).summoner(id);
+            const xpRequired = await rarity.connect(wallet).xp_required(adventurer._level);
+            const className = await rarity.connect(wallet).classes(adventurer._class);
 
             const nextLeveling = new Date(adventurer._log.toNumber() * 1000).toLocaleString()
 
@@ -38,15 +38,17 @@ export async function runParty() {
 
     if (adventurePartyAddress) {
 
-        const adventureParty: AdventureParty = await AdventureParty__factory.connect(adventurePartyAddress, wallet);
+        const adventureParty = <AdventureParty> await ethers.getContractAt("AdventureParty", adventurePartyAddress);
 
         if (await canAdventure()) {
-            let trx = await adventureParty.adventureAll();
+            console.info("Running Adventure...");
+            let trx = await adventureParty.connect(wallet).adventureAll();
             await trx.wait();
         }
 
         if (await canLevelUp()) {
-            let trx = await adventureParty.levelUpAll();
+            console.info("Leveling up...");
+            let trx = await adventureParty.connect(wallet).levelUpAll();
             await trx.wait();
         }
     }
@@ -56,14 +58,14 @@ async function canAdventure() {
     const [wallet] = await ethers.getSigners();
     const adventurePartyAddress = process.env.ADVENTURE_PARTY;
     if (adventurePartyAddress) {
-        const adventureParty: AdventureParty = await AdventureParty__factory.connect(adventurePartyAddress, wallet);
-        const rarity: Rarity = await Rarity__factory.connect(RARITY_CONTRACT, wallet);
+        const adventureParty = <AdventureParty> await ethers.getContractAt("AdventureParty", adventurePartyAddress);
+        const rarity = <Rarity> await ethers.getContractAt("Rarity", RARITY_CONTRACT);
 
-        const adventurersCount = await adventureParty.adventurerCount();
+        const adventurersCount = await adventureParty.connect(wallet).adventurerCount();
 
         for (let i = 0; i < adventurersCount.toNumber(); i++) {
-            const id = await adventureParty.adventurerAt(i);
-            const adventurer = await rarity.summoner(id);
+            const id = await adventureParty.connect(wallet).adventurerAt(i);
+            const adventurer = await rarity.connect(wallet).summoner(id);
 
             const log = adventurer._log.toNumber() * 1000;
             if (log <= (new Date().getTime()) ) {
@@ -79,16 +81,16 @@ async function canLevelUp() {
     const [wallet] = await ethers.getSigners();
     const adventurePartyAddress = process.env.ADVENTURE_PARTY;
     if (adventurePartyAddress) {
-        const adventureParty: AdventureParty = await AdventureParty__factory.connect(adventurePartyAddress, wallet);
-        const rarity: Rarity = await Rarity__factory.connect(RARITY_CONTRACT, wallet);
+        const adventureParty = <AdventureParty> await ethers.getContractAt("AdventureParty", adventurePartyAddress);
+        const rarity = <Rarity> await ethers.getContractAt("Rarity", RARITY_CONTRACT);
 
-        const adventurersCount = await adventureParty.adventurerCount();
+        const adventurersCount = await adventureParty.connect(wallet).adventurerCount();
 
         for (let i = 0; i < adventurersCount.toNumber(); i++) {
-            const id = await adventureParty.adventurerAt(i);
-            const adventurer = await rarity.summoner(id);
+            const id = await adventureParty.connect(wallet).adventurerAt(i);
+            const adventurer = await rarity.connect(wallet).summoner(id);
 
-            const xpRequired = await rarity.xp_required(adventurer._level);
+            const xpRequired = await rarity.connect(wallet).xp_required(adventurer._level);
 
             if (adventurer._xp.gte(xpRequired)) {
                 return true;

